addons/account/models/account_move.py

```diff
@@ -1660,6 +1662,11 @@ def _compute_amount_total_words(self):
         for move in self:
             move.amount_total_words = move.currency_id.amount_to_text(move.amount_total).replace(',', '')
 
+    @api.depends('move_type')
```

Comment:
Either you make it depend on company (so it changes if you change the company).
Or you keep the default and just put it in the lambda/make your own default method

Reply:
True, forgot about company... I can't really use a default, as the move_type won't be assigned at that point